### ISSUE TITLE
`Duplicate sshd config key` failure

### ISSUE THREAD

**Issue by austinbutler** [NixOS/nixpkgs#234683]
### Describe the bug

```
       error:
       Failed assertions:
       - Duplicate sshd config key; does your capitalization match the option's?
```

### Steps To Reproduce
1. Run `nixos-rebuild` with existing working config, see error

### Expected behavior

Able to build successfully or have an easier time figuring out what's wrong.

### Additional context

Looks like this was added recently in https://github.com/NixOS/nixpkgs/pull/231467. I think the error is a bit confusing, and I'm not sure how I'm triggering it. Initially I saw `sshd` so I searched my config for `sshd`: zero matches. Okay... so I [searched NixOS options for `sshd`](https://search.nixos.org/options?channel=unstable&size=50&sort=relevance&type=packages&query=sshd). At the very bottom I found `services.sshd.enable`, which says "Alias of services.openssh.enable." Ah-ha. So now I'm searching for `openssh` in my config. There are two instances. One where I have SSHD enabled and I don't permit root login, the other from home-manager where I allow certain public keys.

```nix
{
  services.openssh = {
    enable = true;
    settings.permitRootLogin = "no";
  };
}
```

```nix
    openssh.authorizedKeys.keyFiles = [./desktop.pub ./phone.pub ./nas.pub ./laptop.pub];
```

Huh, I don't see any duplicate keys there. Now I have to go read the module's source code. I find https://github.com/NixOS/nixpkgs/pull/231467. Read it a bit, still confused. Well that PR mentions https://github.com/NixOS/nixpkgs/pull/193757. I look at the changes there, lots of things just changed the capitalization of options. So I try changing `settings.permitRootLogin` to `settings.PermitRootLogin`, now a rebuild works.

So here's my feedback. The error message is way too terse. Is there a reason there's not a bit more info here? It mentions the _alias_ (`services.sshd`), which on NixOS options search has a single result, and not `services.openssh`, which has 40 results. It mentions a "duplicate" "key", which when we're talking about SSH makes one think of SSH keys. I thought for a moment maybe I'd added the same public key multiple times or something. The only way to know it's a "duplicate" is to read the PR to understand what is being considered a duplicate. So it would also really help if we could add the "duplicate" key to the error output.

Something like this might be better:

```
Duplicate "services.openssh" or "services.sshd" Nix configuration option found: "permitRootLogin".
This can happen when the capitalization of the Nix option, in this case "permitRootLogin",
does not match the capitalization of the sshd_config option. See `man_sshd_config`.
```

### Notify maintainers

- @nyanotech 
- @teto 

### Metadata

```yml
 - system: `"x86_64-linux"`
 - host os: `Linux 6.3.3, NixOS, 23.11 (Tapir), 23.11pre488860.e10802309bf`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(root): `"home-manager, nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by budimanjojo** [comment]
I also got this. Thanks to this issue I now know how it's happening. I also think the error message is a bit too confusing.

**Comment by Shawn8901** [comment]
There is a PR adding the keys to the error message:
https://github.com/NixOS/nixpkgs/pull/234755
Possibly already whats requested :slightly_smiling_face:

**Comment by R-VdP** [comment]
Can this issue be closed since #234755 was merged?
